administration/backpressure.md

```diff
@@ -2,68 +2,119 @@
-
-The setting behaves similarly to the above scenario with `Mem_Buf_Limit` when the non-default `storage.pause_on_chunks_overlimit` is enabled. 
-
-When (default) `storage.pause_on_chunks_overlimit` is disabled, the input will not pause when the memory limit is reached. Instead, it will switch to only buffering logs in the filesystem. The disk spaced used for filesystem buffering can be limited with `storage.total_limit_size`.
-
-See the [Buffering & Storage](buffering-and-storage.md) docs for more information.
-
-## About pause and resume Callbacks
-
-Each plugin is independent and not all of them implements the **pause** and **resume** callbacks. As said, these callbacks are just a notification mechanism for the plugin.
-
-One example of a plugin that implements these callbacks and keeps state correctly is the [Tail Input](../pipeline/inputs/tail.md) plugin. When the **pause** callback is triggered, it pauses its collectors and stops appending data. Upon **resume**, it resumes the collectors and continues ingesting data. Tail will track the current file offset when it pauses and resume at the same position. If the file has not been deleted or moved, it can still be read.
-
-With the default `storage.type memory` and `Mem_Buf_Limit`, the following log messages will be emitted for pause and resume:
-
-```
+It's possible for logs or data to be ingested or created faster than the ability to
+flush it to some destinations. A common scenario is when reading from big log files,
+especially with a large backlog, and dispatching the logs to a backend over the
+network, which takes time to respond. This generates backpressure leading to high
+memory consumption in the service.
+
+To avoid backpressure, Fluent Bit implements a mechanism in the engine that restricts
+the amount of data an input plugin can ingest. Restriction is done through the
+configuration parameters `Mem_Buf_Limit` and `storage.Max_Chunks_Up`.
+
+As described in the [Buffering](../concepts/buffering.md) concepts section, Fluent
+Bit offers two modes for data handling: in-memory only (default) and in-memory +
+filesystem (optional).
+
+The default `storage.type memory` buffer can be restricted with `Mem_Buf_Limit`. If
+memory reaches this limit and you reach a backpressure scenario, you won't be able
+to ingest more data until the data chunks that are in memory can be flushed. The
+input pauses and Fluent Bit
+[emits](https://github.com/fluent/fluent-bit/blob/v2.0.0/src/flb_input_chunk.c#L1334)
+a `[warn] [input] {input name or alias} paused (mem buf overlimit)` log message.
+
+Depending on the input plugin in use, this might lead to discard incoming data (for
+example, TCP input plugin). The tail plugin can handle pause without data loss; it
+stores its current file offset and resumes reading later. When buffer memory is
+available, the input resumes accepting logs. Fluent Bit
```

Comment:
```suggestion
Depending on the input plugin in use, this might cause incoming data to be discarded
(for example, TCP input plugin). The tail plugin can handle pauses without data
ingloss, storing its current file offset and resuming reading later. When buffer
memory is available, the input resumes accepting logs. Fluent Bit
```